src/fmu/dataio/datastructure/meta/content.py

```diff
@@ -181,200 +178,400 @@ class BoundingBox3D(BoundingBox2D):
+
+    description: Optional[List[str]] = Field(default=None)
+    """A list of strings, freetext description of this data, if applicable."""
 
-    # only relevant for grid properties
     geometry: Optional[Geometry] = Field(default=None)
+    """The geometry of the object, i.e. the grid that an object representing a grid
+    property is derivative of. See :class:`Geometry`."""
 
-    # Only valid for coordinate based meta.
     bbox: Optional[Union[BoundingBox3D, BoundingBox2D]] = Field(default=None)
+    """A block containing the bounding box for this data. Only applicable if the
+    object is coordinate-based. See :class:`BoundingBox3D` and :class:`BoudingBox2D`."""
+
+    format: str = Field(examples=["irap_binary"])
+    """A reference to a known file format."""
 
-    description: Optional[List[str]] = Field(
-        default=None,
-    )
-    format: str = Field(
-        examples=["irap_binary"],
-    )
     grid_model: Optional[GridModel] = Field(default=None)
-    is_observation: bool = Field(
-        title="Is observation flag",
-    )
-    is_prediction: bool = Field(
-        title="Is prediction flag",
-    )
+    """A block containing information pertaining to grid model content.
+    See :class:`GridModel`."""
+
+    is_observation: bool
+    """True if this is an observation."""
+
+    is_prediction: bool
+    """True if this is a prediction."""
+
     layout: Optional[enums.Layout] = Field(
         default=None,
         examples=["regular", "cornerpoint"],
     )
-    name: str = Field(
-        description=(
-            "Name of the data object. If stratigraphic, "
-            "match the entry in the stratigraphic column"
-        ),
-        examples=["VIKING GP. Top"],
-    )
-    offset: float = Field(
-        default=0.0,
-        allow_inf_nan=False,
-    )
+    """A reference to the layout of the data object. See :class:`enums.Layout`."""
+
+    offset: float = Field(default=0.0, allow_inf_nan=False)
+    """If a specific horizon is represented with an offset, e.g.
+    "2 m below Top Volantis"."""
+
     spec: Optional[specification.AnySpecification] = Field(default=None)
-    stratigraphic_alias: Optional[List[str]] = Field(default=None)
-    stratigraphic: bool = Field(
-        description=(
-            "True if data object represents an entity in the stratigraphic column"
-        ),
-    )
-    tagname: Optional[str] = Field(
-        default=None,
-        description="A semi-human readable tag for internal usage and uniqueness",
-        examples=["ds_extract_geogrid", "ds_post_strucmod"],
-    )
+    """A block containing the specs for this object, if applicable.
+    See :class:`specification.AnySpecification`."""
+
     time: Optional[Time] = Field(default=None)
+    """A block containing lists of objects describing timestamp information for this
+    data object, if applicable. See :class:`Time`.
+    .. note:: ``data.time`` items can be repeated to include many time stamps."""
```

Comment:
"""A block containing lists of objects describing timestamp information for this
    data object, if applicable, like flow simulator restart dates, or dates for seismic 4D surveys. See :class:`Time`.
    .. note:: ``data.time`` items can currently hold maximum two values."""